beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -583,11 +581,7 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
             // port is removed, which is assumed to be associated with the discv5 protocol (and
             // therefore irrelevant for other libp2p components).
             let mut out_list = enr.multiaddr();
-            out_list.retain(|addr| {
-                addr.iter()
-                    .find(|v| matches!(v, MProtocol::Udp(_)))
-                    .is_none()
-            });
+            out_list.retain(|addr| !addr.iter().any(|v| matches!(v, MProtocol::Udp(_))));
```

Comment:
Clippy's getting pretty clever! This looks right to me but I wouldn't mind some eyes on it from a network dev

Reply:
lgtm